### PR TITLE
feat(onyx-733): design changes in alerts list screen

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-733] 

### Description

Small changes in the alerts screen:

- use StickySubHeader component
- alert title / subtitle in two lines
- gray separator

Demo:

<video src="https://github.com/artsy/eigen/assets/3934579/1f8a8d66-5259-4e04-8f5e-cc053139552f" width="400px;" />

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- design changes in profile alerts list screen - @nickskalkin 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-733]: https://artsyproduct.atlassian.net/browse/ONYX-733?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by olerichter00 — DISMISSED** [review]
Looks good 🌟

**Comment by nickskalkin** [comment]
I've hidden bottom tabs by mistake - going change that

**Review by MounirDhahri — APPROVED** [review]
Clean! thanks

File: docs/add_a_new_screen.md
Comment (MounirDhahri) [inline docs/add_a_new_screen.md:1]: thanks for making these updates

**Review by dariakoko — APPROVED** [review]

**Pull request merged**